packages/react/types/tests.tsx

```diff
@@ -240,6 +240,18 @@ const anim1 = keyframes`
     }
   >['css']
 
+  /*
+     Additional brackets here prevent type reference('EmotionJSX.LibraryManagedAttributes<...>') from being resolved to type alias name('_HasCssPropAsIntended7').
+     Without them, dtslint will compare the expected type to the type name.
+  */
+  type _HasCssPropAsIntended7 = [
+    // $ExpectType { foo: boolean; } | ({ css?: Interpolation<Theme>; } & { foo: number; className: string; })
+    EmotionJSX.LibraryManagedAttributes<
+      {},
+      { foo: number; className: string } | { foo: boolean }
+    >
+  ]
```

Comment:
This is like a unit test. I am more interested in an integration test for this change. How did you come across this to be an issue? Maybe it was related to code like this?
```ts
type DiscriminatedClassNameSupportProp =
  | {
      variant: "foo";
      className?: string;
    }
  | {
      variant: "bar";
    };

function DiscriminatedClassNameSupportComponent(
  props: DiscriminatedClassNameSupportProp,
) {
  return null;
}

<DiscriminatedClassNameSupportComponent
  variant="foo"
  // this should be OK
  css={{ color: "hotpink" }}
/>;

<DiscriminatedClassNameSupportComponent
  variant="bar"
  // this shouldn't be OK
  css={{ color: "hotpink" }}
/>;
```

Reply:
Yes, that was my case. May I add tests like that?

Reply:
Yes. Please replace the tests you have here with something closer to the one above :)

Reply:
Sure. I'll update PR tonight. Thank you for comments!